### PR TITLE
refactor(api-docs): move type filter and remove api: prefix from names

### PR DESCRIPTION
**Pull request by kuangp**
Signed-off-by: Phil Kuang <pkuang@factset.com>

Minor UI tweaks overlooked from #5984
## Hey, I just made a Pull Request!
Screen Shot 2021-06-17 at 1 04 50 PM

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by benjdlambert** [comment]
Hey @kuangp, we had a small issue on master when you made this PR. If you rebase on top it should get you on a stable master.

File: plugins/catalog/src/components/CatalogTable/columns.tsx [hunk 1]
```diff
@@ -19,13 +19,15 @@ import { OverflowTooltip, TableColumn } from '@backstage/core';
 import { Chip } from '@material-ui/core';
 import { EntityRow } from './types';
 
-export function createNameColumn(): TableColumn<EntityRow> {
+export function createNameColumn(
+  defaultKind: string = 'Component',
```
Comment (Rugvip) [inline plugins/catalog/src/components/CatalogTable/columns.tsx:23]: Best make this an options object instead, i.e.

```ts
createNameColumn({
  defaultKind: 'API'
})
```

File: plugins/catalog/src/index.ts [hunk 2]
```diff
@@ -17,7 +17,10 @@
 export * from './components/AboutCard';
 export { CatalogLayout } from './components/CatalogPage';
 export { CatalogResultListItem } from './components/CatalogResultListItem';
-export { CatalogTable } from './components/CatalogTable';
+export {
+  CatalogTable,
+  columnFactories as catalogTableColumnFactories,
```
Comment (Rugvip) [inline plugins/catalog/src/index.ts:22]: Let's use the pattern from [EntityTable](https://github.com/backstage/backstage/blob/809bcd396b80d032547fecac62ae9ba6f04de326/plugins/catalog-react/src/components/EntityTable/EntityTable.tsx#L77) for this instead

**Review by Rugvip — COMMENTED** [review]
Thanks!

**Comment by kuangp** [comment]
@Rugvip made your suggested changes, thanks!

**Review by timbonicus — APPROVED** [review]
Beautiful! 👏

**Pull request merged**